packages/cactus-test-tooling/src/main/typescript/quorum/quorum-mp-test-ledger.ts

```diff
@@ -0,0 +1,244 @@
+    return this._imageTag;
+  }
+
+  public get imageName(): string {
+    return this._imageName;
+  }
+
+  public get imageFqn(): string {
+    return this._imageFqn;
+  }
+
+  public get autoRemove(): boolean {
+    return this._autoRemove;
+  }
+
+  public get emitContainerLogs(): boolean {
+    return this._emitContainerLogs;
+  }
+
+  public async start(omitPull = false): Promise<Container> {
+    const docker = new Dockerode();
+    if (this.containerId.isPresent()) {
+      this.log.warn(`Container ID provided. Will not start new one.`);
+      const container = docker.getContainer(this.containerId.get());
+      return container;
+    }
+    if (!omitPull) {
+      await Containers.pullImage(this.imageFqn);
+    }
+
+    const dockerEnvVars: string[] = new Array(...this.envVars).map(
+      (pairs) => `${pairs[0]}=${pairs[1]}`,
+    );
+
+    const createOptions = {
+      HostConfig: {
+        AutoRemove: this.autoRemove,
+        Env: dockerEnvVars,
+        Privileged: true,
+        PublishAllPorts: true,
+      },
+    };
+
+    this.log.debug(`Starting ${this.imageFqn} with options: `, createOptions);
+
+    return new Promise<Container>((resolve, reject) => {
+      const eventEmitter: EventEmitter = docker.run(
+        this.imageFqn,
+        [],
+        [],
+        createOptions,
+        {},
+        (err: Error) => {
+          if (err) {
+            const errorMessage = `Failed to start container ${this.imageFqn}`;
+            const exception = new RuntimeError(errorMessage, err);
+            this.log.error(exception);
+            reject(exception);
+          }
+        },
+      );
+
+      eventEmitter.once("start", async (container: Container) => {
+        const { id } = container;
+        this.log.debug(`Started ${this.imageFqn} successfully. ID=${id}`);
+        this._containerId = Optional.ofNonNull(id);
+
+        if (this.emitContainerLogs) {
+          const logOptions = { follow: true, stderr: true, stdout: true };
+          const logStream = await container.logs(logOptions);
+          logStream.on("data", (data: Buffer) => {
+            const fnTag = `[${this.imageFqn}]`;
+            this.log.debug(`${fnTag} %o`, data.toString("utf-8"));
+          });
+        }
+
+        try {
+          await Containers.waitForHealthCheck(this.containerId.get());
+          resolve(container);
+        } catch (ex) {
+          this.log.error(ex);
+          reject(ex);
+        }
+      });
+    });
+  }
+
+  public get container(): Optional<Container> {
+    const docker = new Dockerode();
+    return this.containerId.isPresent()
+      ? Optional.ofNonNull(docker.getContainer(this.containerId.get()))
+      : Optional.empty();
+  }
+
+  public async stop(): Promise<unknown> {
+    return Containers.stop(this.container.get());
+  }
+
+  public async destroy(): Promise<unknown> {
+    return this.container.get().remove();
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```

Comment:
@travis-payne Not sure here, but maybe this linter disable line is no longer needed here? If I'm right then please remove it, if it is necessary then a specific justification would be nice as a comment right above it for future reference.

Reply:
Fixed